### PR TITLE
fix(sdk): set log retention for cloud.Function (tf-aws and awscdk)

### PR DESCRIPTION
**Pull request by exoego**
Part of https://github.com/winglang/wing/issues/4134

Added `logRetentionDays` property to `cloud.Function`.
It is implemented for tf-aws and awscdk.

I skipped tf-gcp and tf-azure since the ticket does not mention and other cloud provider appears to have default retention:
- Cloud Logging: 30 days 
- Azure Monitor Application Insights: 90 days

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/target-awscdk/function.ts [hunk 1]
```diff
@@ -36,6 +36,13 @@ export class Function extends cloud.Function implements IAwsFunction {
     // bundled code is guaranteed to be in a fresh directory
     const bundle = createBundle(this.entrypoint);
 
+    const logRetentionDays =
+      props.logRetentionDays === undefined
+        ? 30
+        : props.logRetentionDays < 0
+        ? undefined // Negative value means Infinite retention
```
Comment (tsuf239) [inline libs/wingsdk/src/target-awscdk/function.ts:43]: Shouldn't we allow this as well? (what if a wing user does want infinite retention?)

**Review by tsuf239 — COMMENTED** [review]
Is it possible add a wing test? (is there a way maybe an HTTP call to verify this property was set?)

**Comment by tsuf239** [comment]
also, when I think of it now, maybe it will be good to add a "global" log retention value (let's say I want to set any of my resources to 1 week retention period), which probably should be in a different PR 
@skorfmann WDYT?

File: libs/wingsdk/src/target-awscdk/function.ts (hunk 1 above)
Comment (exoego) [reply libs/wingsdk/src/target-awscdk/function.ts:43]: Yes, it should allow infinite retention.
If a user specifies a negative value (e.g. `-1`), `logRetentionDays` will be `undefined`, which means infinite retention in CloudWatch.

Comment (tsuf239) [reply libs/wingsdk/src/target-awscdk/function.ts:43]: oh I see now, nice!

**Comment by exoego** [comment]
@tsuf239 Rebased into the main

**Review by tsuf239 — APPROVED** [review]
Nice!
image

**Pull request merged**